mani_skill/envs/tasks/tabletop/roll_ball.py

```diff
@@ -0,0 +1,166 @@
+from typing import Any, Dict, Union
+
+import numpy as np
+import torch
+
+import mani_skill.envs.utils.randomization as randomization
+from mani_skill.agents.robots import Fetch, Panda, Xmate3Robotiq
+from mani_skill.envs.sapien_env import BaseEnv
+from mani_skill.sensors.camera import CameraConfig
+from mani_skill.utils import sapien_utils
+from mani_skill.utils.building import actors
+from mani_skill.utils.registration import register_env
+from mani_skill.utils.scene_builder.table import TableSceneBuilder
+from mani_skill.utils.structs.pose import Pose
+from mani_skill.utils.structs.types import Array, GPUMemoryConfig, SimConfig
+from transforms3d.euler import euler2quat
+
+@register_env("RollBall-v1", max_episode_steps=60)
+class RollBallEnv(BaseEnv):
+    """
+    Task Description
+    ----------------
+    A simple task where the objective is to push and roll a ball to a goal region at the other end of the table
+
+    Randomizations
+    --------------
+    - the ball's xy position is randomized on top of a table in the region [0.2, 0.5] x [-0.4, 0.7]. It is placed flat on the table
+    - the target goal region is marked by a red/white circular target. The position of the target is randomized on top of a table in the region [-0.4, -0.7] x [0.2, -0.9]
+
+    Success Conditions
+    ------------------
+    - the ball's xy position is within goal_radius (default 0.1) of the target's xy position by euclidean distance.
+    """
+
+    SUPPORTED_REWARD_MODES = ["dense", "none"]
+    SUPPORTED_ROBOTS = ["panda"]
+
+    agent: Union[Panda]
+
+    goal_radius: float = 0.1 # radius of the goal region
+    ball_radius: float = 0.035 # radius of the ball
+    reward_reach_proportion: float = 0.08 # proportion of the reward that is given for reaching the ball
+
+
+
+    def __init__(self, *args, robot_uids="panda", robot_init_qpos_noise=0.02, **kwargs):
+        self.robot_init_qpos_noise = robot_init_qpos_noise
+        super().__init__(*args, robot_uids=robot_uids, **kwargs)
+
+    @property
+    def _default_sim_config(self):
+        return SimConfig(
+            gpu_memory_cfg=GPUMemoryConfig(
+                found_lost_pairs_capacity=2**25, max_rigid_patch_count=2**18
+            )
+        )
+
```

Comment:
please add a default sensor config. Since this is the same table top u can copy the pick cube.